pvlib/shading.py

```diff
@@ -0,0 +1,193 @@
+
+    .. math::
+
+        \psi(z') = \arctan \left [
+            \frac{(1 - z') \sin \beta}
+                 {\mathrm{GCR}^{-1} + (z' - 1) \cos \beta}
+        \right ]
+
+    Where :math:`GCR = B/C` and :math:`z' = z/B`. The average masking angle
+    :math:`\overline{\psi} = \int_0^1 \psi(z') \mathrm{d}z'` is then
+    evaluated symbolically using Maxima (using :math:`X = 1/\mathrm{GCR}`):
+
+    .. code-block:: none
+
+        load(scifac)    /* for the gcfac function */
+        assume(X>0, cos(beta)>0, cos(beta)-X<0);   /* X is 1/GCR */
+        gcfac(integrate(atan((1-z)*sin(beta)/(X+(z-1)*cos(beta))), z, 0, 1))
+
+    This yields the equation implemented by this function:
+
+    .. math::
+
+        \overline{\psi} = \
+            &-\frac{X}{2} \sin\beta \log | 2 X \cos\beta - (X^2 + 1)| \\
+            &+ (X \cos\beta - 1) \arctan \frac{X \cos\beta - 1}{X \sin\beta} \\
+            &+ (1 - X \cos\beta) \arctan \frac{\cos\beta}{\sin\beta} \\
+            &+ X \log X \sin\beta
+
+    The pvlib-python authors have validated this equation against numerical
+    integration of :math:`\overline{\psi} = \int_0^1 \psi(z') \mathrm{d}z'`.
+
+    References
+    ----------
+    .. [1] D. Passias and B. Källbäck, "Shading effects in rows of solar cell
+       panels", Solar Cells, Volume 11, Pages 281-291.  1984.
+       DOI: 10.1016/0379-6787(84)90017-6
+    """
+    # wrap it in an array so that division by zero is handled well
+    beta = np.radians(np.array(surface_tilt))
+    sin_b = np.sin(beta)
+    cos_b = np.cos(beta)
+    X = 1/gcr
+
+    with np.errstate(divide='ignore', invalid='ignore'):  # ignore beta=0
+        term1 = -X * sin_b * np.log(np.abs(2 * X * cos_b - (X**2 + 1))) / 2
+        term2 = (X * cos_b - 1) * np.arctan((X * cos_b - 1) / (X * sin_b))
+        term3 = (1 - X * cos_b) * np.arctan(cos_b / sin_b)
+        term4 = X * np.log(X) * sin_b
+
+    psi_avg = term1 + term2 + term3 + term4
+    # when beta=0, divide by zero makes psi_avg NaN.  replace with 0:
+    psi_avg = np.where(np.isfinite(psi_avg), psi_avg, 0)
+
+    if isinstance(surface_tilt, pd.Series):
+        psi_avg = pd.Series(psi_avg, index=surface_tilt.index)
+
+    return np.degrees(psi_avg)
+
+
+def sky_diffuse_passias(masking_angle):
+    r"""
+    The diffuse irradiance loss caused by row-to-row sky diffuse shading.
+
+    Even when the sun is high in the sky, a row's view of the sky dome will
+    be partially blocked by the row in front. This causes a reduction in the
+    diffuse irradiance incident on the module. The reduction depends on the
+    masking angle, the elevation angle from a point on the shaded module to
+    the top of the shading row.  SAM assumes the "worst-case" loss where the
+    masking angle is calculated for the bottom of the array [1]_. In [2]_
+    the masking angle is calculated as the average across the module height.
+
+    This function, as in [2]_, makes the assumption that sky diffuse
+    irradiance is isotropic.
```

Comment:
Entirely editorial - I'd swap the order of the last and next-to-last sentences, so that the focus stays on the Passias approach.